### PR TITLE
feat: bignumber support

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/processor/marketplace-events/agreement.ts [hunk 1]
```diff
@@ -15,11 +15,11 @@ const handlers: HandlersObject<MarketplaceEvent, BaseEventProcessorOptions> = {
   async NewAgreement (event: MarketplaceEvent, options: BaseEventProcessorOptions): Promise<void> {
     const newAgreement = event.payload
 
-    await Agreement.upsert(newAgreement) // Agreement might already exist
+    const [agreement] = await Agreement.upsert(newAgreement) // Agreement might already exist
     logger.info(`Created new Agreement with ID ${newAgreement.agreementReference} for offer ${newAgreement.offerId}`)
 
     if (options.manager) {
-      await options.manager.pin(newAgreement.dataReference, parseInt(newAgreement.size))
+      await options.manager.pin(newAgreement.dataReference, agreement.size)
```
Comment (vojtechsimetka) [inline src/processor/marketplace-events/agreement.ts:22]: This has been changed from `newAgreement` to `agreement` . Is that correct?

File: src/utils.ts [hunk 2]
```diff
@@ -30,6 +31,14 @@ import { CliInitDbOptions, JobManagerOptions } from './definitions'
 import { JobsManager } from './jobs-manager'
 import { IpfsProvider } from './providers/ipfs'
 
+export function bn (v: any): BigNumber {
```
Comment (vojtechsimetka) [inline src/utils.ts:34]:
```suggestion
export function bn (v: string | Number | BigNumber): BigNumber {
```

File: src/utils.ts [hunk 3]
```diff
@@ -30,6 +31,14 @@ import { CliInitDbOptions, JobManagerOptions } from './definitions'
...
+  return new BigNumber(v)
+}
+
+export function bnFloor (v: any): BigNumber {
```
Comment (vojtechsimetka) [inline src/utils.ts:38]:
```suggestion
export function bnFloor (v: BigNumber): BigNumber {
```

File: test/unit/cli.spec.ts [hunk 4]
```diff
@@ -251,7 +251,7 @@ describe('CLI', function () {
       process.env.RIFS_DB = dbPath
 
       // Init the DB
-      const sequelize = await sequelizeFactory(dbPath)
+      const sequelize = sequelizeFactory(dbPath)
```
Comment (vojtechsimetka) [inline test/unit/cli.spec.ts:254]: Is this not async function?

File: src/utils.ts (hunk 2 above)
Comment (vojtechsimetka) [inline src/utils.ts:34]: Not a big fan. See https://github.com/rsksmart/rif-marketplace-cache/pull/254#discussion_r469495298

File: src/processor/marketplace-events/agreement.ts (hunk 1 above)
Comment (nduchak) [reply src/processor/marketplace-events/agreement.ts:24]: Nope. It's just remove `parseInt`

File: test/unit/cli.spec.ts (hunk 4 above)
Comment (nduchak) [reply test/unit/cli.spec.ts:254]: Not more

File: src/utils.ts (hunk 2 above)
Comment (nduchak) [reply src/utils.ts]: fixed

**Pull request merged**